app/src/main/java/org/thoughtcrime/securesms/notifications/DefaultMessageNotifier.java

```diff
@@ -324,41 +334,47 @@ private void sendSingleThreadNotification(@NonNull  Context context,
     String                             messageIdTag   = String.valueOf(notifications.get(0).getTimestamp());
 
     NotificationManager notificationManager = ServiceUtil.getNotificationManager(context);
+
+    // Notifications can be bundled together in Android R (API 30) and above - so if that's the case
+    // and we already have
     for (StatusBarNotification notification: notificationManager.getActiveNotifications()) {
       if ( (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R && notification.isAppGroup() == bundled)
               && messageIdTag.equals(notification.getNotification().extras.getString(LATEST_MESSAGE_ID_TAG))) {
         return;
       }
     }
 
+    // Set when this notification occurred so that notifications can be sorted by date
     long timestamp = notifications.get(0).getTimestamp();
     if (timestamp != 0) builder.setWhen(timestamp);
 
     builder.putStringExtra(LATEST_MESSAGE_ID_TAG, messageIdTag);
 
     CharSequence text = notifications.get(0).getText();
+    CharSequence body = MentionUtilities.highlightMentions(
+              text != null ? text : "",
+              false,
+              false,
+              true, // no styling here, only text formatting
+              notifications.get(0).getThreadId(),
+              context
+    );
 
     builder.setThread(notifications.get(0).getRecipient());
     builder.setMessageCount(notificationState.getMessageCount());
 
+
+
     // TODO: Removing highlighting mentions in the notification because this context is the libsession one which
     // TODO: doesn't have access to the `R.attr.message_sent_text_color` and `R.attr.message_received_text_color`
     // TODO: attributes to perform the colour lookup. Also, it makes little sense to highlight the mentions using
     // TODO: the app theme as it may result in insufficient contrast with the notification background which will
     // TODO: be using the SYSTEM theme.
     builder.setPrimaryMessageBody(recipient,
                                   notifications.get(0).getIndividualRecipient(),
-                                  text == null ? "" : text,
+                                  text != null ? body : "",
```

Comment:
if a exists then use b?

Reply:
No - if we have text then use the mention-substituted text, otherwise use an empty string.

This is all WIP & in cleanup anyway.